### ISSUE TITLE
Support using @param {type} to replace @is[NumTag]

### ISSUE THREAD

**Issue by jackey8616** [lukeautry/tsoa#707]
Wish we can full use of `@param` tag in jsDoc comment.  
  
## Sorting
- **I'm submitting a ...**
  - [x] feature request

- I confirm that I
  - [x] used the [search](https://github.com/lukeautry/tsoa/search?type=Issues) to make sure that a similar issue hasn't already been submit

## Expected Behavior
```ts
  /**
   * @param {Long} index
   */
```

## Current Behavior
```ts
  /**
   * @isLong index
   */
```

## Detailed Description
Since that the JsDoc provides a `@param`  tag is able to follow with type,  
I think it would be good if we combined `@param` and tsoa's self-defined `@is[NumType]` tag together.  
It can reduce the complexity of method comment to prevent such situation: 
```ts
/**
  * @isLong index1
  * @isInt index2
  * @isDouble index3
  * @param {number} index1 - index1 description
  * @param {number} index2 - index2 description
  * @param {number} index3 - index3 description
  */
```

About the type of `long`, `int`, `double` type is not define in JsDoc,  
I think we can declare global type in tsoa to make a shadow copy like below: 
```ts
declare global {
  type Long = number;
  type Double = number;
  type Float = number;
  ...
}
```
## Breaking change?
Yes, for `@is[NumType]` tag.  
We can simply deprecated the number tag and add some hint while generation.  
And after few versions, then full remove support of `@is[NumType]` tag.  
The generation detail is still the old logic which use in number type tag, but just replace the `@is[NumType]` tag to `@param` tag.

**Comment by WoH** [comment]
The first question I'd have is how we could handle jsDoc annotations on classes or interfaces.

i.e. 
```ts
interface NumericTypes {
  double: number,
  /**
  * @isInt 
  */
  int: number,

  /**
  * @isLong 
  */
  long: number
}
```

**Comment by jackey8616** [comment]
I think `@type` tag can do the job.
[JsDoc @Type](https://jsdoc.app/tags-type.html)

**Comment by WoH** [comment]
I can see why you'd want that, but I don't want to end up with a scenario where someone uses `@type` and a type annotation in their code.

Is this really so bothersome to users if they can just define 
```ts
/**
 * @isLong
 */
export type Long = number;
```

and use that all throughout the code.

I'm not saying `@param {Long} index` is worse than `@isLong`, but fixing this has consequences where I don't see the clear upside any more when considering the entire picture.
But I'd love to hear more opinions here, thanks for bringing it up.

**Comment by jackey8616** [comment]
@WoH 
TL;DR, make only `@param {type}` supports for methods, and keep `@is[NumType]` for models.

I agree with simple `@isLong` comment in single variable or interface.  

The idea of replacing `@is[NumType]` tag actually comes from some usecase like below:
```ts
/**
 * @param {number} param_one Description
 * @param {number} param_two Description
 * @isInt param_one
 * @isDouble param_two
 */
public async someMethod(
  @Query() param_one: number,
  @Query() param_two: number,
): Promise<void> { return; }
```

Once the parameters is over than an amount( like four or five), supplying both parameter type and description will make the comment too wordy.

While I'm trying to implement this feature, I do notice that `@param` is not suitable for models fields.  
If I try to make it fit for models, that would not so gentle.

**Comment by WoH** [comment]
> The idea of replacing @is[NumType] tag actually comes from some usecase like below:

I think the pattern we should promote here is use `@isX` as rarely as possible.
In your example, I'd highly recommend to use `@isX` on a type alias that you can then reuse throughout your program. Similar to [this `Word` type](https://tsoa-community.github.io/docs/upgrading.html#new-features).

```ts
// src/tsoa-types.ts
/**
 * @isInt
 */
export type Int = number;

/**
 * @isLong
 */
export type Long = number;

// controller
import {Int, Long} from '../tsoa-types.ts'

/**
 * @param param_one Description
 * @param param_two Description
 */
public async someMethod(
  @Query() param_one: Int,
  @Query() param_two: Long,
): Promise<void> { return; }
```

**Comment by jackey8616** [comment]
You are right, I misused `@is` comment tag.  
I think the solution you provide is more better and gentle.  
Suggest to put in tsoa's doc.  

I will close this issue, and maybe make some contribute to tsoa's doc.  
Thank you all, @WoH .

**Comment by WoH** [comment]
I'd love to have more observability, i tried to link the example I used in the getting started docs but couldn't find it the first time, which clearly is a bad sign :laughing: 
I use many types like [this](https://tsoa-community.github.io/docs/descriptions.html#model-descriptions) at work so we get a nice combination of reusable comments/examples/validation through a simple reusable OpenAPI component.
